st2common/st2common/models/api/base.py

```diff
@@ -171,22 +171,43 @@ def callfunction(*args, **kwargs):
             types = copy.copy(arg_types)
             more = [args.pop(0)]
 
+            def cast_value(value_type, value):
+                if value_type == bool:
+                    def cast_func(value):
+                        return value.lower() in ['1', 'true']
+                else:
+                    cast_func = value_type
+
+                result = cast_func(value)
+                return result
+
             if types:
                 argspec = inspect.getargspec(f)
                 names = argspec.args[1:]
 
-                for name in names:
```

Comment:
This approach with trying to consume non-keyword arguments first was very fragile.

It only really worked correctly if method didn't take more than a single parameter. If method took more than one parameter (e.g. `get_all` in this PR), we needed to abuse `kwargs` to make it work.

The problem with abusing kwargs is that it breaks whole "arg_types" functionality / abstraction and user can also pass arbitrary unsupported parameters to the API and we will just simply swallow them and ignore it (or in some cases where we pass kwargs directly to the underlying methods, things will explode in a weird manner).